iter_lint/range_len_fix.py

```diff
@@ -1,54 +1,34 @@
-                                                                   iter=Call(
-                    func=enumerate_node, args=[node_iterable], keywords=[]),
-                    body=node.body,
-                    orelse=node.orelse), node))
-            return node
-
-        @staticmethod
-        def is_range_len(node: ast.For):
-            try:
-                logging.debug("check")
-                logging.debug(pformat(node))
-                return node.iter.func.id == 'range' and node.iter.args[0].func.id == 'len'
-            except AttributeError as e:
-                logging.debug("attr error!" + str(e))
-                return False
 
     def check_for_smell(self) -> bool:
-        """Check if the smell occurs between `starting_line` and `end_line` in `source_code`"""
-        return self.EnumerateFixer(change_node=False).visit(ast.parse(self.source_code))
+        return bool(re.search(self.source_code, r"\n*( *|\t*)for (\w+) in range\(len\((.+)\)\):"))
 
     def fix_smell(self) -> str:
-        """Return a fixed version of the code without the code smell"""
-        return astor.to_source(self.EnumerateFixer(change_node=True).visit(ast.parse(self.source_code)))
-
-
-def swap_enumerate(code: str):
-    return EnumerateFixer().visit(ast.parse(code))
+        fixed_source = self.source_code.rstrip() + os.linesep
+        for for_match in re.finditer(r"(?=\n*|\r*)( *|\t*)for (\w+) in range\(len\((.+)\)\):", self.source_code):  # https://regex101.com/r/ib52OT/1
+            indent, index, seq = for_match.groups()
+            for_statement = indent + "for " + re.escape(index) + " in range(len(" + re.escape(seq) + ")):" + os.linesep
+            content_rgx = indent + "for " + re.escape(index) + r" in range\(len\(" + re.escape(seq) + r"\)\):" + os.linesep + "((" + indent + r"(\s+).+" + os.linesep + ")*)"
+            content = re.search(content_rgx, fixed_source).group(1)
+            assign = re.search(r"([^\s]+)\s*=\s*(.*)(" + re.escape(seq) + r"\[" + re.escape(index) + r"\]" + r")(.*)", content)  # https://regex101.com/r/CnibMS/1
+            elm = "elm"
+            i = 1
+            while re.search(elm, fixed_source):
+                elm = "elm" + str(i)
+            fixed_content = content
+            if assign and assign.group(2).strip() == assign.group(4).strip() == "":
+                elm = assign.group(1)
+                fixed_content = content.replace(assign.group(0), "")
+                fixed_content = os.linesep.join([line for line in fixed_content.splitlines() if line.strip() != ""])
+            fixed_content = fixed_content.replace(seq + "[" + index + "]", elm)
+            if fixed_content.strip() == "":
+                fixed_content += indent + "    pass"
+            fixed_content = "{}for {}, {} in enumerate({}):".format(indent, index, elm, seq) + os.linesep + fixed_content
```

Comment:
We're running py3.6+ - use f-strings!
```suggestion
            fixed_content = f"{indent}for {index}, {elm} in enumerate({seq}):" + os.linesep + fixed_content
```